GetStream/stream-chat-android
# [CAS-855] Refactor MessageOptionsConfig

Refactor xml attributes applying, `MessageOptionsView.Config` and feature flags logic in `MessageListView`

### 🎯 Goal
Firstly we want to support dynamic updates of channel::config updates, but currently it isn't supported by BE.
We should correctly define available options in the message options screen. Their availability is logical `AND` between XML attributes and channel config.

### 🛠 Implementation details

1. Remove all UI attributes from `MessageOptionsView.Config` to `MessageListViewStyle`
2. Keep in `MessageOptionsView.Config` only feature flags and create instance only when message options is being shown
3. Add logic of feature flag calculation to operator `MessageOptionsView.Config::invoke`

### 🧪 Testing

1. Update dashboard channel config
2. Reload channel
3. Check options are updated accordingly

### ☑️ Checklist
